internal/storage/auth/keycloak/user_test.go

```diff
@@ -297,3 +297,68 @@ func TestKeyCloakUserRepo_UserToKeyCloakUser(t *testing.T) {
 		assert.Equal(t, "123456", (*got.Attributes)["employee_id"][0])
 	})
 }
+
+func TestKeyCloakUserRepo_GetAll(t *testing.T) {
+	t.Run("should get all users successfully", func(t *testing.T) {
+		// given
+		identityConfig := suite.IdentityConfig(t, context.Background())
+		userRepo := NewUserRepository(identityConfig)
+		user1 := &entities.User{
+			Username:    "user1",
+			FirstName:   "John",
+			LastName:    "Doe",
+			Email:       "john.doe@green-ecolution.de",
+			EmployeeID:  "EMP001",
+			PhoneNumber: "+49 123456789",
+		}
+		user2 := &entities.User{
+			Username:    "user2",
+			FirstName:   "Jane",
+			LastName:    "Doe",
+			Email:       "jane.doe@green-ecolution.de",
+			EmployeeID:  "EMP002",
+			PhoneNumber: "+49 987654321",
+		}
+
+		suite.EnsureUserExists(t, user1)
+		suite.EnsureUserExists(t, user2)
+
+		// when
+		users, err := userRepo.GetAll(context.Background())
+
+		// then
+		assert.NoError(t, err)
+		assert.NotNil(t, users)
+		assert.GreaterOrEqual(t, len(users), 2)
+		assert.True(t, containsUser(users, *user1), "user1 should be in the list")
```

Comment:
I am not sure, but can you maybe use `assert.Contain`?

Reply:
That's the reason why I implemented the containsUser function. The assert.Contains method checks if the pointer (memory reference) is included or not, which doesn't work in my case